cpp/src/Slice/Parser.cpp

```diff
@@ -2485,17 +2485,14 @@ Slice::ClassDef::createDataMember(
         }
     }
 
-    SyntaxTreeBasePtr dlt = defaultValueType;
-    string dv = defaultValueString;
-
-    if (dlt || (dynamic_pointer_cast<Enum>(type) && !dv.empty()))
+    if (defaultValueString)
     {
         // Validate the default value.
-        if (!validateConstant(name, type, dlt, dv, false))
+        if (!validateConstant(name, type, defaultValueType, *defaultValueString, false))
```

Comment:
defaultValueType is "in-out". validateConstant can replace a nullptr by an non-null EnumeratorPtr.